CONTRIBUTING.md

```diff
@@ -15,7 +15,7 @@ Please note that this project is released with a [Contributor Code of Conduct][c
 0. Configure and install the dependencies: `bundle install`
 0. Make sure the tests pass on your machine: `bundle exec rspec spec`
 0. Create a new branch: `git checkout -b my-branch-name`
-0. Make your change, add tests, and make sure the tests still pass
+0. Make your change, add tests, and make sure the tests still pass and that no warnings are raised
```

Comment:
This has me stumped. I can't seem to figure out how to make the suite fail when there are warnings outputted. 

I guess another route to go would be to add rubocop. I created a branch where I added rubocop and was able to bring it down to ~270 violations. Most of those remaining could be fixed either by overriding rubocop's default rules or via code changes (line length in particular is probably the most prominent violation).